### DISCUSSION TITLE
Is there any way to legalize macros?

### DISCUSSION THREAD

**Discussion by jkim971201** [The-OpenROAD-Project/OpenROAD-flow-scripts#1576] [Q&A]
Hi, I want to legalize macros to see how RePlAce-MS works well.
In my knowledge, mpl and mpl2 determine the positions of macros.
I just need to remove the overlaps between them, not to place them.
(I checked that dpl can not legalize macro cells.)

Thanks.

**Comment by maliberty** [comment]
No, we assume that if you place macros manually you will do so legally.

**Reply by RaidToRadar** [reply]
Then by any chance, is there a recommended legalization script that can be used externally from ORFS? I am currently experimenting with using a external MP solution but want to make it compatible with  ORFS. Thou it seems to not generate legal solutions.

**Reply by maliberty** [reply]
You could write one but I don't know of one offhand.
